chapters/classes.tex

```diff
@@ -344,18 +344,14 @@ \subsection{Component Variability Prefixes discrete, parameter, constant}\label{
 \item
   A variable \lstinline!vc! declared with the \lstinline!parameter!\indexinline{parameter} prefix remains constant during transient analysis, with a value determined by the initialization problem.
 \item
-  A \emph{discrete-time} variable \lstinline!vd! has a vanishing time derivative between events.
-  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere,
-  as the derivative is not even defined at the events, and it is not legal
-  to apply \lstinline!der! to discrete-time variables as they are not continuous. During transient analysis the variable
-  can only change its value at event
-  instants (see \cref{events-and-synchronization}).
+  A \emph{discrete-time} variable \lstinline!vd! is a variable declared with the \lstinline!discrete!\indexinline{discrete} prefix, or a variable without any prefix that is not a continuous-time variable.
+  It has a vanishing time derivative between events.
+  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere, as the derivative is not even defined at the events, and it is not legal to apply \lstinline!der! to discrete-time variables as they are not continuous.
```

Comment:
I find the motivation for not allowing `der` strange as it would apply just as fine to our continuous-time variables (which in general may even have discontinuities between events).  I doubt that there actually exist good reasons to not allow `der` for the discrete-time variables, and whatever we say should be harmonized with the definition of `der(expr)`:
> The time derivative of `expr`. If the expression `expr` is a scalar it needs to be a subtype of `Real`. The expression and all its time-varying subexpressions must be continuous and semi-differentiable.

Our discrete-time variables are clearly semi-differentiable, so the least we should do here is to not try to give any reason at all:
```suggestion
  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere, as the derivative is not even defined at the events.
  It is not allowed to apply \lstinline!der! to discrete-time variables.
```

However, this rule is so strange that we might as well consider dropping it right now.  From a technical point of view, applying `der` to a discrete-time variable is a breeze compared to applying it to some semi-differentiable `noEvent` (or `max`, for that matter) mess.  In particular, dealing with `der` applied to a discrete-time variable during index reduction would be a simple matter of removing the variable from the analysis, similar to a constant or parameter.

Reply:
Sure, `max` with just two arguments isn't that bad.

My main point is still that as long as we allow `der` to be applied to continuous-time semi-differentiable expressions, it's just an arbitrary restriction to not allow it for discrete-time expressions (and variables).